File: java/maxProfit.java
class MaxProfit {
    public static void main(String[] args) {
        int[] prices = {7, 1, 5, 3, 6, 4};
        System.out.println(maxProfit(prices));
    }

    public static int maxProfit(int[] prices) {
        int minPrice = Integer.MAX_VALUE;
        int maxProfit = 0;

        for(int i =0; i<prices.length; i++){
            if(prices[i] < minPrice){
                minPrice = prices[i];
            }
            if(prices[i] - minPrice > maxProfit){
                maxProfit = prices[i] - minPrice;
            }        
        }
        return maxProfit;
    }
}

File: js/maxProfit.js
var maxProfit =  function(prices) {
    let max = 0;
    let minPrice = prices[0];
    
    for(let day of prices){
        const dayToBuy = day;

        const dayToSell = dayToBuy - minPrice;
        max = Math.max(dayToSell, max);
        minPrice = Math.min(minPrice, dayToBuy);
    }
    return max;
}

console.log(maxProfit([7,1,5,3,6,4]));